### PR TITLE
feat: Store push token in UserDefaults

### PR DESCRIPTION
**Pull request by KaterinaWire**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

We don't need to store `pushToken` in the `UserClient` property because this only applies to `selfUser`. We will use the new user defaults to store the push token locally.

Needs releases with:

- [ wireapp/wire-ios-data-model/pull/1363 ] 

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by KaterinaWire** [comment]
> some SwiftLint warnings ⚠️

Fixed

**Review by David-Henner — APPROVED** [review]

File: Source/SessionManager/SessionManager.swift [hunk 1]
```diff
@@ -774,17 +774,29 @@ public final class SessionManager: NSObject, SessionManagerType {
         registerObservers(account: account, session: userSession)
     }
 
-    private func updateOrMigratePushToken(session userSession: ZMUserSession) {
-        if let currentToken = userSession.selfUserClient?.pushToken?.tokenType,
-            currentToken != requiredPushTokenType {
-            pushLog.safePublic("deleting current token")
+    func updateOrMigratePushToken(session userSession: ZMUserSession) {
+        guard let localToken = PushTokenStorage.pushToken else {
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Source/SessionManager/SessionManager.swift:778]: Yes, but then we will have 2 push token sources: `UserDefaults` and `UserClient.pushToken`.

[the remark this answers is not in the document]
Comment (johnxnguyen) [reply Source/SessionManager/SessionManager.swift:778]: Or we could even move the token from the self user client to the push token storage with a (hot fix) patch.

[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Source/SessionManager/SessionManager.swift]: Implemented

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**